operator/standalone/create.go

```diff
@@ -0,0 +1,236 @@
+package standalone
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	pipeline "github.com/ccremer/go-command-pipeline"
+	helmv1beta1 "github.com/crossplane-contrib/provider-helm/apis/release/v1beta1"
+	crossplanev1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/vshn/appcat-service-postgresql/apis/postgresql/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CreateStandalonePipeline is a pipeline that creates a new instance in the target deployment namespace.
+// Currently, it's optimized for first-time creation scenarios and may fail when reconciling existing instances.
+type CreateStandalonePipeline struct {
+	operatorNamespace string
+	client            client.Client
+
+	// TODO: Idea: Maybe store and retrieve the following fields from the context for safe access. This would require some convenience getters and setters though.
+
+	instance   *v1alpha1.PostgresqlStandalone
+	config     *v1alpha1.PostgresqlStandaloneOperatorConfig
+	helmValues HelmValues
+	helmChart  *v1alpha1.ChartMeta
+}
+
+// NewCreateStandalonePipeline creates a new pipeline with the required dependencies.
+func NewCreateStandalonePipeline(client client.Client, instance *v1alpha1.PostgresqlStandalone, operatorNamespace string) *CreateStandalonePipeline {
+	return &CreateStandalonePipeline{
+		instance:          instance,
+		client:            client,
+		operatorNamespace: operatorNamespace,
+	}
+}
+
+// RunPipeline executes the pipeline with configured business logic steps.
+// This should only be executed once per pipeline as it stores intermediate results in the struct.
+func (p *CreateStandalonePipeline) RunPipeline(ctx context.Context) error {
+	return pipeline.NewPipeline().
```

Comment:
I like this pipeline design. It's similar to what we've done in Lieutenant. 👍 

But, do all these functions have to be exported? As far as I can see they are only used within this package. Is this a limitation of this pipeline framework?

Reply:
Yes, it's very much inspired by this :)

No, not a limitation. I don't have a real reason to export it. Except maybe to make it clear that it's an important business function and codeclimate reminds us to add some minimal godoc? I don't really have hard feeling.